vcd/resource_vcd_vapp_vm.go

```diff
@@ -1607,13 +1732,19 @@ func resourceVcdVAppVmUpdateExecute(d *schema.ResourceData, meta interface{}, ex
 	// this represents fields which have to be changed in cold (with VM power off)
 	if d.HasChanges("cpu_cores", "power_on", "disk", "expose_hardware_virtualization", "boot_image",
 		"hardware_version", "os_type", "description", "cpu_hot_add_enabled",
-		"memory_hot_add_enabled") || memoryNeedsColdChange || cpusNeedsColdChange || networksNeedsColdChange {
-
-		log.Printf("[TRACE] VM %s has changes: memory(%t), cpus(%t), cpu_cores(%t), power_on(%t), disk(%t), expose_hardware_virtualization(%t),"+
-			" boot_image(%t), hardware_version(%t), os_type(%t), description(%t), cpu_hot_add_enabled(%t), memory_hot_add_enabled(%t), network(%t)",
-			vm.VM.Name, d.HasChange("memory"), d.HasChange("cpus"), d.HasChange("cpu_cores"), d.HasChange("power_on"), d.HasChange("disk"),
-			d.HasChange("expose_hardware_virtualization"), d.HasChange("boot_image"), d.HasChange("hardware_version"),
-			d.HasChange("os_type"), d.HasChange("description"), d.HasChange("cpu_hot_add_enabled"), d.HasChange("memory_hot_add_enabled"), d.HasChange("network"))
+		"memory_hot_add_enabled", "firmware", "boot_options.0.efi_secure_boot") || memoryNeedsColdChange || cpusNeedsColdChange || networksNeedsColdChange {
```

Comment:
Does this nested element path gets evaluated correctly? I remember there were some issues on getting that value previously.

Reply:
It does, here is a sample debug output when only changing efi secure boot
```
[TRACE] VM TestAccVcdVmAdvancedComputeProperties_empty-19433 has changes: memory(false), cpus(false),
 cpu_cores(false),power_on(false), disk(false), expose_hardware_virtualization(false), boot_image(false), 
hardware_version(false), os_type(false), description(false),cpu_hot_add_enabled(false), 
memory_hot_add_enabled(false), firmware(false),efi_secure_boot(true) network(false): 
timestamp=2023-09-05T18:52:53.845+0300
```

Reply:
Great, thanks!